vm/src/stdlib/select.rs

```diff
@@ -0,0 +1,75 @@
+use crate::function::OptionalOption;
+use crate::pyobject::{PyObjectRef, PyResult};
+use crate::vm::VirtualMachine;
+
+#[cfg(unix)]
+fn select_select(
+    rlist: PyObjectRef,
+    wlist: PyObjectRef,
+    xlist: PyObjectRef,
+    timeout: OptionalOption<f64>,
+    vm: &VirtualMachine,
+) -> PyResult<PyObjectRef> {
+    use nix::sys::select;
+    use nix::sys::time::{TimeVal, TimeValLike};
+    use std::os::unix::io::RawFd;
+
+    let seq2set = |list| -> PyResult<(Vec<RawFd>, select::FdSet)> {
+        let v = vm.extract_elements(list)?;
+        let mut fds = select::FdSet::new();
+        for fd in &v {
+            fds.insert(*fd);
+        }
+        Ok((v, fds))
+    };
+
+    let (rlist, mut r) = seq2set(&rlist)?;
+    let (wlist, mut w) = seq2set(&wlist)?;
+    let (xlist, mut x) = seq2set(&xlist)?;
+
+    let mut timeout = timeout
+        .flat_option()
+        .map(|to| TimeVal::nanoseconds((to * 1e9) as i64));
+
+    select::select(
+        None,
+        Some(&mut r),
+        Some(&mut w),
+        Some(&mut x),
+        timeout.as_mut(),
+    )
+    .map_err(|err| super::os::convert_nix_error(vm, err))?;
```

Comment:
We should check here if the select has failed due to an interrupt and if so we should check signals and continue. I would suggest just adding a `TODO` for now.